spec/models/case/sar/offender_complaint_spec.rb

```diff
@@ -0,0 +1,703 @@
+        kase = build(:offender_sar_complaint, number_final_pages: -562)
+        expect(kase).not_to be_valid
+        expect(kase.errors[:number_final_pages]).to eq ['must be a positive whole number']
+      end
+    end
+
+    context 'valid values' do
+      it 'is valid when string represents a positive whole number' do
+        kase = build(:offender_sar_complaint, number_final_pages: 4835)
+        expect(kase).to be_valid
+      end
+    end
+  end
+
+  describe 'third party details' do
+    describe 'third_party_names' do
+      it 'validates third party names when third party is true' do
+        kase = build :offender_sar_complaint, :third_party, third_party_name: '', third_party_company_name: ''
+        expect(kase).not_to be_valid
+        expect(kase.errors[:third_party_name]).to eq ["can't be blank if company name not given"]
+        expect(kase.errors[:third_party_company_name]).to eq ["can't be blank if representative name not given"]
+      end
+
+      it 'validates third party names when recipient is third party' do
+        kase = build :offender_sar_complaint, third_party: false, third_party_name: '',
+                      third_party_company_name: '', recipient: 'third_party_recipient'
+        expect(kase).not_to be_valid
+        expect(kase.errors[:third_party_name]).to eq ["can't be blank if company name not given"]
+        expect(kase.errors[:third_party_company_name]).to eq ["can't be blank if representative name not given"]
+      end
+
+      it 'does not validate third_party names when ecipient is not third party too' do
+        kase = build :offender_sar_complaint, third_party: false, third_party_name: '',
+                      third_party_company_name: '', recipient: 'subject_recipient'
+        expect(kase).to be_valid
+      end
+    end
+
+
+    describe 'third party relationship' do
+      it 'must be present when thrid party is true' do
+        kase = build :offender_sar_complaint, third_party: true, third_party_relationship: ''
+        expect(kase).not_to be_valid
+        expect(kase.errors[:third_party_relationship]).to eq ["can't be blank"]
+      end
+
+      it 'must be present when third party is false but recipient is third party' do
+        kase = build :offender_sar_complaint, third_party: false, third_party_relationship: '',
+                      recipient: 'third_party_recipient'
+        expect(kase).not_to be_valid
+        expect(kase.errors[:third_party_relationship]).to eq ["can't be blank"]
+      end
+
+      it 'does not validates presence of third party relationship when recipient is not third party' do
+        kase = build :offender_sar_complaint, third_party: false, third_party_relationship: '',
+                      recipient: 'subject_recipient'
+        expect(kase).to be_valid
+      end
+    end
+  end
+
+  describe '#subject' do
+    it 'is the same as subject full name' do
+      kase = create :offender_sar_complaint
```

Comment:
Another possible small refactor here - I think a lot of these could be `build_stubbed` rather than create as we are testing in memory object validations.